pkg/exec/exec.go

```diff
@@ -17,14 +17,43 @@ type ExecClient interface {
 }
 
 // ExecuteCommand executes the given command in the pod's container
-func ExecuteCommand(client ExecClient, compInfo common.ComponentInfo, command []string, show bool) (err error) {
-	reader, writer := io.Pipe()
+func ExecuteCommand(client ExecClient, compInfo common.ComponentInfo, command []string, show bool, consoleOutputStdout *io.PipeWriter, consoleOutputStderr *io.PipeWriter) (err error) {
```

Comment:
This is definitely something to consider: of course this is definitely down to individual opinion (a 'heuristic'), but to me this doesn't yet cross that line of 'this should be merged into a struct of params', though it is definitely very close to that line. I had considered merging the stderr/stdout into a struct, since they are semantically related, but it didn't feel like enough of an improvement (6 params -> 5) to be worth doing, and likewise none of the other parameters appeared to be semantically related enough to combine.

Reply:
Makes sense.